hi-ml-azure/src/health_azure/utils.py

```diff
@@ -895,54 +896,125 @@ def _log_conda_dependencies_stats(conda: CondaDependencies, message_prefix: str)
         logging.debug(f"    {p}")
 
 
-def _retrieve_unique_deps(dependencies: List[str], keep_method: str = "first") -> List[str]:
+def _split_dependency(dep_str: str) -> Tuple[str, ...]:
+    """Splits a string like those coming from PIP constraints into 3 parts: package name, operator, version.
+    The operator field can be empty if no constraint is found at all
+
+    :param dep_str: A pip constraint string, like "package-name>=1.0.1"
+    :return: A tuple of [package name, operator, version ]
+    """
+    parts: List[str] = re.split('(<=|==|=|>=|<|>)', dep_str)
+    if len(parts) == 1:
+        return (parts[0].strip(), "", "")
+    if len(parts) == 3:
+        return tuple(p.strip() for p in parts)
+    raise ValueError(f"Unable to split this package string: {dep_str}")
+
+
+class PackageDependency:
+    def __init__(self, dependency_str: str) -> None:
+        self.package_name = ""
+        self.operator = ""
+        self.version = ""
+        self._split_dependency_str(dependency_str)
+
+    def _split_dependency_str(self, dependency_str: str) -> None:
+        parts = _split_dependency(dependency_str)
+        self.package_name = parts[0]
+        if len(parts) > 0:
```

Comment:
I think this should be len(parts) > 1?

Reply:
and for the sake of paranoia, also guard line 926 similarly

Reply:
Wait, no. all of these checks are obsolete because split... returns a 3-tuple in all cases

Reply:
Good catch - I've removed this check